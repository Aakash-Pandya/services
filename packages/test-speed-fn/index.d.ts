export function testSpeedFn(str: string): Promise<object>
export function webPageTest(str: string): Promise<void>
