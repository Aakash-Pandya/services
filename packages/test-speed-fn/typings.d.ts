export function testSpeedFn(str: string): Promise<string>
