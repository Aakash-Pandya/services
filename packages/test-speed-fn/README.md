# test-url

It expects having https://github.com/marcelduran/webpagetest-api installed globally.
