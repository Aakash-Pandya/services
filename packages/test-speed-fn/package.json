{
  "name": "test-speed-fn",
  "scripts": {
    "out": "tsc -p .",
    "in": "tsc -w",
    "test": "jest",
    "prove": "node ./prove.js"
  },
  "typings": "./index.d.ts",
  "main": "./src/test-speed-fn.js",
  "version": "0.1.0",
  "dependencies": {
    "env-fn": "2.1.0",
    "helpers-fn": "^0.4.1",
    "opener": "^1.5.1",
    "psi": "4.0.1",
    "rambdax": "3.7.0"
  },
  "devDependencies": {
    "@babel/core": "7.10.2",
    "@babel/plugin-proposal-object-rest-spread": "7.10.1",
    "@babel/preset-env": "7.10.2",
    "@types/jest": "25.2.3",
    "jest": "26.0.1",
    "typescript": "3.9.3"
  },
  "jest": {
    "testEnvironment": "node",
    "testRegex": ".*\\.spec\\.js$",
    "collectCoverageFrom": [
      "src/*.js"
    ]
  },
  "repository": "https://github.com/selfrefactor/services/tree/master/packages/test-speed",
  "license": "MIT",
  "author": "Dejan Toteff <deyan8284@gmail.com>",
  "description": "test-speed",
  "private": false
}
