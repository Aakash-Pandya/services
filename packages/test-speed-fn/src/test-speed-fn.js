const { envFn } = require('env-fn')
envFn('special')
const { gtmetrix } = require('./modules/gtmetrix')
const { gtmetrixPoll } = require('./modules/gtmetrix-poll')
const { log } = require('helpers-fn')
const { psi } = require('./modules/psi')
const { webPageTest } = require('./modules/web-page-test')

async function testGTMetrix(url){
  const gtmetrixPollCommand = await gtmetrix(url)
  const gtmetrixResult = await gtmetrixPoll(gtmetrixPollCommand)
  log(gtmetrixResult, 'obj')
}

async function testPSI(url){
  const psiResult = await psi(url)
  log(psiResult, 'obj')
}

async function testSpeedFn(url){
  // const [gtmetrixPollCommand, psiResult, _] = await Promise.all([
  //   gtmetrixModule(url),
  //   psiModule(url),
  //   webPageTest(url),
  // ])
  // log(psi, 'obj')
  // const gtmetrixResult = await gtmetrixPoll(gtmetrixPollCommand as string)
  // log(gtmetrixResult, 'obj')
}

exports.testGTMetrix = testGTMetrix
exports.testPSI = testPSI
exports.testSpeedFn = testSpeedFn
