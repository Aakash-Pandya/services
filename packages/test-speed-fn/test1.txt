  FIRST_CONTENTFUL_PAINT_MS: {
    percentile: 4587,
    distributions: [ [Object], [Object], [Object] ],
    category: 'SLOW'
  },
  CUMULATIVE_LAYOUT_SHIFT_SCORE: {
    percentile: 84,
    distributions: [ [Object], [Object], [Object] ],
    category: 'SLOW'
  },
  FIRST_INPUT_DELAY_MS: {
    percentile: 72,
    distributions: [ [Object], [Object], [Object] ],
    category: 'FAST'
  },
  LARGEST_CONTENTFUL_PAINT_MS: {
    percentile: 6065,
    distributions: [ [Object], [Object], [Object] ],
    category: 'SLOW'
  }