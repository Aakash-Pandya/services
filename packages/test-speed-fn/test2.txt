{
  audits: {
    'unminified-javascript': {
      id: 'unminified-javascript',
      title: 'Minify JavaScript',
      description: 'Minifying JavaScript files can reduce payload sizes and script parse time. [Learn more](https://web.dev/unminified-javascript).',
      score: 1,
      scoreDisplayMode: 'numeric',
      details: [Object],
      warnings: [],
      numericValue: 0
    },
    'uses-rel-preconnect': {
      id: 'uses-rel-preconnect',
      title: 'Preconnect to required origins',
      description: 'Consider adding `preconnect` or `dns-prefetch` resource hints to establish early connections to important third-party origins. [Learn more](https://web.dev/uses-rel-preconnect).',
      score: 1,
      scoreDisplayMode: 'numeric',
      details: [Object],
      warnings: [],
      numericValue: 0
    },
    'speed-index': {
      id: 'speed-index',
      title: 'Speed Index',
      description: 'Speed Index shows how quickly the contents of a page are visibly populated. [Learn more](https://web.dev/speed-index).',
      score: 0.01,
      scoreDisplayMode: 'numeric',
      displayValue: '16.8 s',
      numericValue: 16758.5
    },
    'largest-contentful-paint-element': {
      id: 'largest-contentful-paint-element',
      title: 'Largest Contentful Paint element',
      description: 'This is the element that was identified as the Largest Contentful Paint. [Learn More](https://web.dev/lighthouse-largest-contentful-paint)',
      score: null,
      scoreDisplayMode: 'informative',
      displayValue: '1 element found',
      details: [Object]
    },
    'uses-passive-event-listeners': {
      id: 'uses-passive-event-listeners',
      title: 'Uses passive listeners to improve scrolling performance',
      description: "Consider marking your touch and wheel event listeners as `passive` to improve your page's scroll performance. [Learn more](https://web.dev/uses-passive-event-listeners).",
      score: 1,
      scoreDisplayMode: 'binary',
      details: [Object]
    },
    'uses-rel-preload': {
      id: 'uses-rel-preload',
      title: 'Preload key requests',
      description: 'Consider using `<link rel=preload>` to prioritize fetching resources that are currently requested later in page load. [Learn more](https://web.dev/uses-rel-preload).',
      score: 1,
      scoreDisplayMode: 'numeric',
      details: [Object],
      numericValue: 0
    },
    'offscreen-images': {
      id: 'offscreen-images',
      title: 'Defer offscreen images',
      description: 'Consider lazy-loading offscreen and hidden images after all critical resources have finished loading to lower time to interactive. [Learn more](https://web.dev/offscreen-images).',
      score: 1,
      scoreDisplayMode: 'numeric',
      details: [Object],
      warnings: [],
      numericValue: 0
    },
    'critical-request-chains': {
      id: 'critical-request-chains',
      title: 'Avoid chaining critical requests',
      description: 'The Critical Request Chains below show you what resources are loaded with a high priority. Consider reducing the length of chains, reducing the download size of resources, or deferring the download of unnecessary resources to improve page load. [Learn more](https://web.dev/critical-request-chains).',
      score: null,
      scoreDisplayMode: 'informative',
      displayValue: '14 chains found',
      details: [Object]
    },
    'user-timings': {
      id: 'user-timings',
      title: 'User Timing marks and measures',
      description: "Consider instrumenting your app with the User Timing API to measure your app's real-world performance during key user experiences. [Learn more](https://web.dev/user-timings).",
      score: null,
      scoreDisplayMode: 'notApplicable',
      details: [Object]
    },
    'total-blocking-time': {
      id: 'total-blocking-time',
      title: 'Total Blocking Time',
      description: 'Sum of all time periods between FCP and Time to Interactive, when task length exceeded 50ms, expressed in milliseconds. [Learn more](https://web.dev/lighthouse-total-blocking-time).',
      score: 0.01,
      scoreDisplayMode: 'numeric',
      displayValue: '2,330 ms',
      numericValue: 2334.5
    },
    'third-party-summary': {
      id: 'third-party-summary',
      title: 'Reduce the impact of third-party code',
      description: 'Third-party code can significantly impact load performance. Limit the number of redundant third-party providers and try to load third-party code after your page has primarily finished loading. [Learn more](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/loading-third-party-javascript/).',
      score: 0,
      scoreDisplayMode: 'binary',
      displayValue: 'Third-party code blocked the main thread for 580 ms',
      details: [Object]
    },
    'unused-css-rules': {
      id: 'unused-css-rules',
      title: 'Remove unused CSS',
      description: 'Remove dead rules from stylesheets and defer the loading of CSS not used for above-the-fold content to reduce unnecessary bytes consumed by network activity. [Learn more](https://web.dev/unused-css-rules).',
      score: 0.5,
      scoreDisplayMode: 'numeric',
      displayValue: 'Potential savings of 125 KB',
      details: [Object],
      numericValue: 750
    },
    'cumulative-layout-shift': {
      id: 'cumulative-layout-shift',
      title: 'Cumulative Layout Shift',
      description: 'Cumulative Layout Shift measures the movement of visible elements within the viewport. [Learn more](https://web.dev/cls).',
      score: 0.54,
      scoreDisplayMode: 'numeric',
      displayValue: '0.234',
      details: [Object],
      numericValue: 0.23433992513020835
    },
    'screenshot-thumbnails': {
      id: 'screenshot-thumbnails',
      title: 'Screenshot Thumbnails',
      description: 'This is what the load of your site looked like.',
      score: null,
      scoreDisplayMode: 'informative',
      details: [Object]
    },
    'server-response-time': {
      id: 'server-response-time',
      title: 'Initial server response time was short',
      description: 'Keep the server response time for the main document short because all other requests depend on it. [Learn more](https://web.dev/time-to-first-byte).',
      score: 1,
      scoreDisplayMode: 'binary',
      displayValue: 'Root document took 100 ms',
      details: [Object],
      numericValue: 98.046
    },
    'first-meaningful-paint': {
      id: 'first-meaningful-paint',
      title: 'First Meaningful Paint',
      description: 'First Meaningful Paint measures when the primary content of a page is visible. [Learn more](https://web.dev/first-meaningful-paint).',
      score: 0,
      scoreDisplayMode: 'numeric',
      displayValue: '17.5 s',
      numericValue: 17508.5
    },
    'layout-shift-elements': {
      id: 'layout-shift-elements',
      title: 'Avoid large layout shifts',
      description: 'These DOM elements contribute most to the CLS of the page.',
      score: null,
      scoreDisplayMode: 'informative',
      displayValue: '4 elements found',
      details: [Object]
    },
    'largest-contentful-paint': {
      id: 'largest-contentful-paint',
      title: 'Largest Contentful Paint',
      description: 'Largest Contentful Paint marks the time at which the largest text or image is painted. [Learn More](https://web.dev/lighthouse-largest-contentful-paint)',
      score: 0,
      scoreDisplayMode: 'numeric',
      displayValue: '31.5 s',
      numericValue: 31545
    },
    'uses-optimized-images': {
      id: 'uses-optimized-images',
      title: 'Efficiently encode images',
      description: 'Optimized images load faster and consume less cellular data. [Learn more](https://web.dev/uses-optimized-images).',
      score: 0.2,
      scoreDisplayMode: 'numeric',
      displayValue: 'Potential savings of 596 KB',
      details: [Object],
      warnings: [],
      numericValue: 3300
    },
    'uses-text-compression': {
      id: 'uses-text-compression',
      title: 'Enable text compression',
      description: 'Text-based resources should be served with compression (gzip, deflate or brotli) to minimize total network bytes. [Learn more](https://web.dev/uses-text-compression).',
      score: 1,
      scoreDisplayMode: 'numeric',
      details: [Object],
      numericValue: 0
    },
    'total-byte-weight': {
      id: 'total-byte-weight',
      title: 'Avoid enormous network payloads',
      description: 'Large network payloads cost users real money and are highly correlated with long load times. [Learn more](https://web.dev/total-byte-weight).',
      score: 0,
      scoreDisplayMode: 'numeric',
      displayValue: 'Total size was 11,902 KB',
      details: [Object],
      numericValue: 12187943
    },
    diagnostics: {
      id: 'diagnostics',
      title: 'Diagnostics',
      description: 'Collection of useful page vitals.',
      score: null,
      scoreDisplayMode: 'informative',
      details: [Object]
    },
    'first-contentful-paint': {
      id: 'first-contentful-paint',
      title: 'First Contentful Paint',
      description: 'First Contentful Paint marks the time at which the first text or image is painted. [Learn more](https://web.dev/first-contentful-paint).',
      score: 0,
      scoreDisplayMode: 'numeric',
      displayValue: '16.8 s',
      numericValue: 16758.5
    },
    'font-display': {
      id: 'font-display',
      title: 'Ensure text remains visible during webfont load',
      description: 'Leverage the font-display CSS feature to ensure text is user-visible while webfonts are loading. [Learn more](https://web.dev/font-display).',
      score: 0,
      scoreDisplayMode: 'binary',
      details: [Object],
      warnings: []
    },
    'max-potential-fid': {
      id: 'max-potential-fid',
      title: 'Max Potential First Input Delay',
      description: 'The maximum potential First Input Delay that your users could experience is the duration of the longest task. [Learn more](https://web.dev/lighthouse-max-potential-fid).',
      score: 0,
      scoreDisplayMode: 'numeric',
      displayValue: '1,360 ms',
      numericValue: 1357
    },
    'no-document-write': {
      id: 'no-document-write',
      title: 'Avoids `document.write()`',
      description: 'For users on slow connections, external scripts dynamically injected via `document.write()` can delay page load by tens of seconds. [Learn more](https://web.dev/no-document-write).',
      score: 1,
      scoreDisplayMode: 'binary',
      details: [Object]
    },
    'mainthread-work-breakdown': {
      id: 'mainthread-work-breakdown',
      title: 'Minimize main-thread work',
      description: 'Consider reducing the time spent parsing, compiling and executing JS. You may find delivering smaller JS payloads helps with this. [Learn more](https://web.dev/mainthread-work-breakdown)',
      score: 0.14,
      scoreDisplayMode: 'numeric',
      displayValue: '7.0 s',
      details: [Object],
      numericValue: 7049.459999999985
    },
    'network-server-latency': {
      id: 'network-server-latency',
      title: 'Server Backend Latencies',
      description: "Server latencies can impact web performance. If the server latency of an origin is high, it's an indication the server is overloaded or has poor backend performance. [Learn more](https://hpbn.co/primer-on-web-performance/#analyzing-the-resource-waterfall).",
      score: null,
      scoreDisplayMode: 'informative',
      displayValue: '0 ms',
      details: [Object],
      numericValue: 0
    },
    'dom-size': {
      id: 'dom-size',
      title: 'Avoids an excessive DOM size',
      description: 'A large DOM will increase memory usage, cause longer [style calculations](https://developers.google.com/web/fundamentals/performance/rendering/reduce-the-scope-and-complexity-of-style-calculations), and produce costly [layout reflows](https://developers.google.com/speed/articles/reflow). [Learn more](https://web.dev/dom-size).',
      score: 1,
      scoreDisplayMode: 'numeric',
      displayValue: '410 elements',
      details: [Object],
      numericValue: 410
    },
    'render-blocking-resources': {
      id: 'render-blocking-resources',
      title: 'Eliminate render-blocking resources',
      description: 'Resources are blocking the first paint of your page. Consider delivering critical JS/CSS inline and deferring all non-critical JS/styles. [Learn more](https://web.dev/render-blocking-resources).',
      score: 0.58,
      scoreDisplayMode: 'numeric',
      displayValue: 'Potential savings of 600 ms',
      details: [Object],
      numericValue: 600
    },
    'network-requests': {
      id: 'network-requests',
      title: 'Network Requests',
      description: 'Lists the network requests that were made during page load.',
      score: null,
      scoreDisplayMode: 'informative',
      details: [Object]
    },
    'uses-long-cache-ttl': {
      id: 'uses-long-cache-ttl',
      title: 'Serve static assets with an efficient cache policy',
      description: 'A long cache lifetime can speed up repeat visits to your page. [Learn more](https://web.dev/uses-long-cache-ttl).',
      score: 0,
      scoreDisplayMode: 'numeric',
      displayValue: '28 resources found',
      details: [Object],
      numericValue: 11169475.504166666
    },
    'bootup-time': {
      id: 'bootup-time',
      title: 'Reduce JavaScript execution time',
      description: 'Consider reducing the time spent parsing, compiling, and executing JS. You may find delivering smaller JS payloads helps with this. [Learn more](https://web.dev/bootup-time).',
      score: 0.36,
      scoreDisplayMode: 'numeric',
      displayValue: '4.6 s',
      details: [Object],
      numericValue: 4601.511999999997
    },
    'efficient-animated-content': {
      id: 'efficient-animated-content',
      title: 'Use video formats for animated content',
      description: 'Large GIFs are inefficient for delivering animated content. Consider using MPEG4/WebM videos for animations and PNG/WebP for static images instead of GIF to save network bytes. [Learn more](https://web.dev/efficient-animated-content)',
      score: 1,
      scoreDisplayMode: 'numeric',
      details: [Object],
      numericValue: 0
    },
    redirects: {
      id: 'redirects',
      title: 'Avoid multiple page redirects',
      description: 'Redirects introduce additional delays before the page can be loaded. [Learn more](https://web.dev/redirects).',
      score: 1,
      scoreDisplayMode: 'numeric',
      displayValue: 'Potential savings of 780 ms',
      details: [Object],
      numericValue: 780
    },
    'uses-responsive-images': {
      id: 'uses-responsive-images',
      title: 'Properly size images',
      description: 'Serve images that are appropriately-sized to save cellular data and improve load time. [Learn more](https://web.dev/uses-responsive-images).',
      score: 0,
      scoreDisplayMode: 'numeric',
      displayValue: 'Potential savings of 5,172 KB',
      details: [Object],
      warnings: [],
      numericValue: 26970
    },
    'final-screenshot': {
      id: 'final-screenshot',
      title: 'Final Screenshot',
      description: 'The last screenshot captured of the pageload.',
      score: null,
      scoreDisplayMode: 'informative',
      details: [Object]
    },
    'uses-webp-images': {
      id: 'uses-webp-images',
      title: 'Serve images in next-gen formats',
      description: 'Image formats like JPEG 2000, JPEG XR, and WebP often provide better compression than PNG or JPEG, which means faster downloads and less data consumption. [Learn more](https://web.dev/uses-webp-images).',
      score: 0,
      scoreDisplayMode: 'numeric',
      displayValue: 'Potential savings of 5,475 KB',
      details: [Object],
      warnings: [],
      numericValue: 29100
    },
    'performance-budget': {
      id: 'performance-budget',
      title: 'Performance budget',
      description: 'Keep the quantity and size of network requests under the targets set by the provided performance budget. [Learn more](https://developers.google.com/web/tools/lighthouse/audits/budgets).',
      score: null,
      scoreDisplayMode: 'notApplicable'
    },
    'unminified-css': {
      id: 'unminified-css',
      title: 'Minify CSS',
      description: 'Minifying CSS files can reduce network payload sizes. [Learn more](https://web.dev/unminified-css).',
      score: 0.88,
      scoreDisplayMode: 'numeric',
      displayValue: 'Potential savings of 19 KB',
      details: [Object],
      numericValue: 150
    },
    'unused-javascript': {
      id: 'unused-javascript',
      title: 'Remove unused JavaScript',
      description: 'Remove unused JavaScript to reduce bytes consumed by network activity. [Learn more](https://web.dev/remove-unused-code/).',
      score: 0,
      scoreDisplayMode: 'numeric',
      displayValue: 'Potential savings of 2,169 KB',
      details: [Object],
      numericValue: 11250
    },
    metrics: {
      id: 'metrics',
      title: 'Metrics',
      description: 'Collects all available metrics.',
      score: null,
      scoreDisplayMode: 'informative',
      details: [Object],
      numericValue: 25818
    },
    interactive: {
      id: 'interactive',
      title: 'Time to Interactive',
      description: 'Time to interactive is the amount of time it takes for the page to become fully interactive. [Learn more](https://web.dev/interactive).',
      score: 0.01,
      scoreDisplayMode: 'numeric',
      displayValue: '25.8 s',
      numericValue: 25818
    },
    'network-rtt': {
      id: 'network-rtt',
      title: 'Network Round Trip Times',
      description: "Network round trip times (RTT) have a large impact on performance. If the RTT to an origin is high, it's an indication that servers closer to the user could improve performance. [Learn more](https://hpbn.co/primer-on-latency-and-bandwidth/).",
      score: null,
      scoreDisplayMode: 'informative',
      displayValue: '0 ms',
      details: [Object],
      numericValue: 0
    },
    'first-contentful-paint-3g': {
      id: 'first-contentful-paint-3g',
      title: 'First Contentful Paint (3G)',
      description: 'First Contentful Paint 3G marks the time at which the first text or image is painted while on a 3G network. [Learn more](https://developers.google.com/web/tools/lighthouse/audits/first-contentful-paint).',
      score: 0,
      scoreDisplayMode: 'numeric',
      displayValue: '36708.5 ms',
      numericValue: 36708.5
    },
    'main-thread-tasks': {
      id: 'main-thread-tasks',
      title: 'Tasks',
      description: 'Lists the toplevel main thread tasks that executed during page load.',
      score: null,
      scoreDisplayMode: 'informative',
      details: [Object]
    },
    'estimated-input-latency': {
      id: 'estimated-input-latency',
      title: 'Estimated Input Latency',
      description: 'Estimated Input Latency is an estimate of how long your app takes to respond to user input, in milliseconds, during the busiest 5s window of page load. If your latency is higher than 50 ms, users may perceive your app as laggy. [Learn more](https://web.dev/estimated-input-latency).',
      score: 0,
      scoreDisplayMode: 'numeric',
      displayValue: '570 ms',
      numericValue: 570.4000000000001
    },
    'resource-summary': {
      id: 'resource-summary',
      title: 'Keep request counts low and transfer sizes small',
      description: 'To set budgets for the quantity and size of page resources, add a budget.json file. [Learn more](https://developers.google.com/web/tools/lighthouse/audits/budgets).',
      score: null,
      scoreDisplayMode: 'informative',
      displayValue: '63 requests • 11,902 KB',
      details: [Object]
    },
    'timing-budget': {
      id: 'timing-budget',
      title: 'Timing budget',
      description: 'Set a timing budget to help you keep an eye on the performance of your site. Performant sites load fast and respond to user input events quickly. [Learn more](https://developers.google.com/web/tools/lighthouse/audits/budgets).',
      score: null,
      scoreDisplayMode: 'notApplicable'
    },
    'first-cpu-idle': {
      id: 'first-cpu-idle',
      title: 'First CPU Idle',
      description: "First CPU Idle marks the first time at which the page's main thread is quiet enough to handle input.  [Learn more](https://web.dev/first-cpu-idle).",
      score: 0,
      scoreDisplayMode: 'numeric',
      displayValue: '23.9 s',
      numericValue: 23943
    }
  },
  metrics: {
    id: 'metrics',
    title: 'Metrics',
    description: 'Collects all available metrics.',
    score: null,
    scoreDisplayMode: 'informative',
    details: { type: 'debugdata', items: [Array] },
    numericValue: 25818
  },
  diagnostics: {
    id: 'diagnostics',
    title: 'Diagnostics',
    description: 'Collection of useful page vitals.',
    score: null,
    scoreDisplayMode: 'informative',
    details: { items: [Array], type: 'debugdata' }
  },
  interactive: {
    id: 'interactive',
    title: 'Time to Interactive',
    description: 'Time to interactive is the amount of time it takes for the page to become fully interactive. [Learn more](https://web.dev/interactive).',
    score: 0.01,
    scoreDisplayMode: 'numeric',
    displayValue: '25.8 s',
    numericValue: 25818
  },
  'first-contentful-paint': {
    id: 'first-contentful-paint',
    title: 'First Contentful Paint',
    description: 'First Contentful Paint marks the time at which the first text or image is painted. [Learn more](https://web.dev/first-contentful-paint).',
    score: 0,
    scoreDisplayMode: 'numeric',
    displayValue: '16.8 s',
    numericValue: 16758.5
  },
  'network-requests': {
    id: 'network-requests',
    title: 'Network Requests',
    description: 'Lists the network requests that were made during page load.',
    score: null,
    scoreDisplayMode: 'informative',
    details: { headings: [Array], items: [Array], type: 'table' }
  },
  'resource-summary': {
    id: 'resource-summary',
    title: 'Keep request counts low and transfer sizes small',
    description: 'To set budgets for the quantity and size of page resources, add a budget.json file. [Learn more](https://developers.google.com/web/tools/lighthouse/audits/budgets).',
    score: null,
    scoreDisplayMode: 'informative',
    displayValue: '63 requests • 11,902 KB',
    details: { items: [Array], headings: [Array], type: 'table' }
  },
  'third-party-summary': {
    id: 'third-party-summary',
    title: 'Reduce the impact of third-party code',
    description: 'Third-party code can significantly impact load performance. Limit the number of redundant third-party providers and try to load third-party code after your page has primarily finished loading. [Learn more](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/loading-third-party-javascript/).',
    score: 0,
    scoreDisplayMode: 'binary',
    displayValue: 'Third-party code blocked the main thread for 580 ms',
    details: {
      summary: [Object],
      items: [Array],
      headings: [Array],
      type: 'table'
    }
  },
  'unused-javascript': {
    id: 'unused-javascript',
    title: 'Remove unused JavaScript',
    description: 'Remove unused JavaScript to reduce bytes consumed by network activity. [Learn more](https://web.dev/remove-unused-code/).',
    score: 0,
    scoreDisplayMode: 'numeric',
    displayValue: 'Potential savings of 2,169 KB',
    details: {
      headings: [Array],
      items: [Array],
      type: 'opportunity',
      overallSavingsBytes: 2220993,
      overallSavingsMs: 11250
    },
    numericValue: 11250
  },
  'speed-index': {
    id: 'speed-index',
    title: 'Speed Index',
    description: 'Speed Index shows how quickly the contents of a page are visibly populated. [Learn more](https://web.dev/speed-index).',
    score: 0.01,
    scoreDisplayMode: 'numeric',
    displayValue: '16.8 s',
    numericValue: 16758.5
  },
  'user-timings': {
    id: 'user-timings',
    title: 'User Timing marks and measures',
    description: "Consider instrumenting your app with the User Timing API to measure your app's real-world performance during key user experiences. [Learn more](https://web.dev/user-timings).",
    score: null,
    scoreDisplayMode: 'notApplicable',
    details: { type: 'table', items: [], headings: [] }
  },
  'first-meaningful-paint': {
    id: 'first-meaningful-paint',
    title: 'First Meaningful Paint',
    description: 'First Meaningful Paint measures when the primary content of a page is visible. [Learn more](https://web.dev/first-meaningful-paint).',
    score: 0,
    scoreDisplayMode: 'numeric',
    displayValue: '17.5 s',
    numericValue: 17508.5
  },
  timing: 12622.97
}