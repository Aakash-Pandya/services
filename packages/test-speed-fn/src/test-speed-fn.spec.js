import { testSpeedFn } from './test-speed-fn'
jest.setTimeout(180000)

const url = 'https://bg.helpkarma.com'

test('happy', async () => {
  const result = await testSpeedFn(url)
  expect(result).toMatchSnapshot()
})
 