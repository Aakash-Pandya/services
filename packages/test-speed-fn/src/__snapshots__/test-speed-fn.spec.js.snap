// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Object {
  "gtmetrix": Object {
    "domInteractive": 1058,
    "firstPaint": 1829,
    "fullyLoaded": 2908,
    "onLoad": 1058,
    "pageLoadTime": 1058,
    "pagespeedScore": 20,
    "yslowScore": 88,
  },
  "lighthouse": Object {
    "blockingResources": Object {
      "displayValue": "Potential savings of 600 ms",
      "score": 0.58,
    },
    "bytes": Object {
      "displayValue": "Total size was 12,219 KB",
      "score": 0,
    },
    "dom": Object {
      "displayValue": "410 elements",
      "score": 1,
    },
    "firstPaint": Object {
      "displayValue": "16.7 s",
      "score": 0,
    },
    "firstPaint3G": Object {
      "displayValue": "34600.5 ms",
      "score": 0,
    },
    "images": Object {
      "displayValue": "Potential savings of 596 KB",
      "score": 0.22,
    },
    "interactive": Object {
      "displayValue": "25.0 s",
      "score": 0.01,
    },
    "largestPaint": Object {
      "displayValue": "30.8 s",
      "score": 0,
    },
    "layoutShiftElements": "5 elements found",
    "loadingExperience": Object {
      "details": Object {
        "firstPaint": Object {
          "score": "slow",
          "value": 4500,
        },
        "inputDelay": Object {
          "score": "fast",
          "value": 22,
        },
        "largestPaint": Object {
          "score": "slow",
          "value": 5407,
        },
        "lauoutShift": Object {
          "score": "slow",
          "value": 81,
        },
      },
      "overallScore": "slow",
      "scoreCounter": Object {
        "fast": 1,
        "slow": 3,
      },
    },
    "originLoadingExperience": Object {
      "details": Object {
        "firstPaint": Object {
          "score": "slow",
          "value": 4500,
        },
        "inputDelay": Object {
          "score": "fast",
          "value": 22,
        },
        "largestPaint": Object {
          "score": "slow",
          "value": 5407,
        },
        "lauoutShift": Object {
          "score": "slow",
          "value": 81,
        },
      },
      "overallScore": "slow",
      "scoreCounter": Object {
        "fast": 1,
        "slow": 3,
      },
    },
    "performanceScore": 0.03,
    "requestsSummary": "62 requests • 12,219 KB",
    "responsiveImages": Object {
      "displayValue": "Potential savings of 5,371 KB",
      "score": 0,
    },
    "server": "Initial server response time was short",
    "spped": Object {
      "displayValue": "15.8 s",
      "score": 0.01,
    },
    "thirdParty": Object {
      "displayValue": "Third-party code blocked the main thread for 760 ms",
      "score": 0,
    },
    "timing": 11998.93,
    "unused": Object {
      "displayValue": "Potential savings of 2,168 KB",
      "score": 0,
    },
  },
}
`;
